src/rust/engine/process_execution/src/docker.rs

```diff
@@ -24,8 +24,9 @@ pub(crate) const SANDBOX_PATH_IN_CONTAINER: &str = "/pants-sandbox";
 pub(crate) const NAMED_CACHES_PATH_IN_CONTAINER: &str = "/pants-named-caches";
 pub(crate) const IMMUTABLE_INPUTS_PATH_IN_CONTAINER: &str = "/pants-immutable-inputs";
 
-/// `CommandRunner` executes processes using a local Docker client.
+/// `CommandRunner` that executes processes using a local Docker client.
 pub struct CommandRunner {
+  inner: Box<dyn crate::CommandRunner>,
```

Comment:
`cache::CommandRunner` uses `Arc` instead?

Reply:
That is to allow for the backtracking stacks, because the same instance ends up wrapped in a cache in one case, and not wrapped in a cache in another.

In this case `Box<dyn>` is fine.